File: Java/Problem_002.java
public class Main {
    public static void main(String[] args) {
        final long limit = 4_000_000L;
        long total = 0;
        long a = 2, b = 8;

        while (a <= limit) {
            total += a;
            long next = 4 * b + a;
            a = b;
            b = next;
        }

        System.out.println(total); // 4613732
    }
}

File: CPP/Problem_002.cpp
#include <iostream>
using namespace std;

int main() {
    const long long limit = 4'000'000;
    long long total = 0;
    long long a = 2, b = 8;

    while (a <= limit) {
        total += a;
        long long next = 4 * b + a;
        a = b;
        b = next;
    }

    cout << total << "\n"; // 4613732
    return 0;
}
